FirebaseCore/Tests/Unit/Swift/FirebaseAppTests.swift

```diff
@@ -0,0 +1,344 @@
+// Copyright 2020 Google
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import XCTest
+@testable import FirebaseCore
+
+private extension Constants {
+  static let testAppName1 = "test_app_name_1"
+  static let testAppName2 = "test_app_name_2"
+}
+
+class FirebaseAppTests: XCTestCase {
+  override func setUp() {
+    super.setUp()
+  }
+
+  override func tearDown() {
+    super.tearDown()
+    FirebaseApp.resetApps()
+  }
+
+  func testSwiftFlagWithSwift() {
+    XCTAssertTrue(FirebaseApp.firebaseUserAgent().contains("swift"))
+  }
+
+  func testConfigure() throws {
+    expectAppConfigurationNotification(appName: Constants.App.defaultName, isDefaultApp: true)
+
+    let configurationAttempt = {
+      try ExceptionCatcher.catchException {
+        FirebaseApp.configure()
+      }
+    }
+    XCTAssertNoThrow(try configurationAttempt())
+
+    let app = try XCTUnwrap(FirebaseApp.app(), "Failed to unwrap default app")
+    XCTAssertEqual(app.name, Constants.App.defaultName)
+    XCTAssertEqual(app.options.clientID, Constants.Options.clientID)
+    XCTAssertEqual(FirebaseApp.allApps?.count, 1)
+
+    // TODO: check registered libraries instances available
```

Comment:
Is this still relevant? What needs to be checked do you think?

Reply:
I noticed that in the [objc tests we check](https://github.com/firebase/firebase-ios-sdk/blob/65e0a9896b5e33cfb82b674070ae27411b1106e8/FirebaseCore/Tests/Unit/FIRAppTest.m#L109) that the newly configured app's registered libraries instances are available. 
It seemed a bit more involved to figure out how to do this in swift and I can leave (or take out!) the `TODO` based on whether or not is it worth testing in the swift tests in the future. This was briefly addressed above [here](https://github.com/firebase/firebase-ios-sdk/pull/5678#discussion_r430426971)